pyfstat/core.py

```diff
@@ -1636,38 +1659,34 @@ def get_semicoherent_twoF(
         #                                       self.BSGLSetup)
         #    return log10_BSGL/np.log10(np.exp(1))
 
-        detStat = 0
+        det_stat_per_segment = self._get_per_segment_det_stat()
+        detStat = det_stat_per_segment.sum()
         if record_segments:
-            self.detStat_per_segment = []
-
-        self.windowRange.tau = int(self.Tcoh)  # TYPE UINT4
-        for tstart in self.tboundaries[:-1]:
-            d_detStat = self._get_per_segment_det_stat(tstart)
-            detStat += d_detStat
-            if record_segments:
-                self.detStat_per_segment.append(d_detStat)
+            self.detStat_per_segment = det_stat_per_segment
 
         return detStat
 
-    def _get_per_segment_det_stat(self, tstart):
-        self.windowRange.t0 = int(tstart)  # TYPE UINT4
-
+    def _get_per_segment_det_stat(self):
         FS = lalpulsar.ComputeTransientFstatMap(
-            self.FstatResults.multiFatoms[0], self.windowRange, False
+            self.FstatResults.multiFatoms[0], self._semicoherent_window_trick, False
         )
 
         if self.BSGL is False:
-            d_detStat = 2 * FS.F_mn.data[0][0]
+            d_detStat = 2 * FS.F_mn.data[:, 0]
         else:
             FstatResults_single = copy.copy(self.FstatResults)
             FstatResults_single.lenth = 1
             FstatResults_single.data = self.FstatResults.multiFatoms[0].data[0]
             FS0 = lalpulsar.ComputeTransientFstatMap(
```

Comment:
It's unrelated to your change, but since you're touching this part anyway, could you add a one-liner comment above this line please:
```
# FIXME: this currently hard-codes BSGL for exactly 2 detectors.
```
Increases the chance that I will at some point fix it. ;)